platform/android/export/export.cpp

```diff
@@ -1975,8 +1976,9 @@ class EditorExportPlatformAndroid : public EditorExportPlatform {
 
 			if (file.ends_with(".so")) {
 				bool enabled = false;
-				for (int i = 0; i < enabled_abis.size(); ++i) {
-					if (file.begins_with("lib/" + enabled_abis[i] + "/")) {
+				for (int i = 0; i < invalid_abis.size(); ++i) {
+					if (file.begins_with("lib/" + invalid_abis[i] + "/")) {
```

Comment:
While they are semantically the same, for clarity I'd use `enabled_abis` for the `for` loop size and for the file check:
```
for (int i = 0; i < enabled_abis.size(); ++i) {
    if (file.begins_with("lib/" + enabled_abis[i] + "/")) {
        invalid_abis.remove(i);
        ...
    }
}
```

This conveys that the logic is going through all the enabled abis and removing the valid ones from the invalid abis list.
It's also safer since the `invalid_abis.remove(i)` call can modify the value returned by `invalid_abis.size()`.

Reply:
In this case we should use `invalid_abis.erase(enabled_abis[i])`? 
```
for (int i = 0; i < enabled_abis.size(); ++i) {
    if (file.begins_with("lib/" + enabled_abis[i] + "/")) {
        invalid_abis.erase(enabled_abis[i]);
        ...
    }
}
```
With `invalid_abis.remove(i)` I'm afraid the index might not match afterwards because the invalid_abis vector will resize.

Alternatively I could use a bool array or bit array to mark the valid abis but when it comes to outputting the error string it becomes quite messy. Something like this 
```
//Checking valid abis
{
	String unsupported_arch = "";
	for (int i = 0; i < enabled_abis.size(); i++) {
		if (!valid_abis[i]) 
			unsupported_arch += enabled_abis[i] + ", ";
	}
	if (unsupported_arch != "") {
		EditorNode::add_io_error("lib for the selected architectures " + unsupported_arch + "is not found in the export template.\
		\nPlease uncheck the unsupported architectures in the export settings or add the required libs in the export template.");
		return ERR_FILE_NOT_FOUND;
	}
}
```

Reply:
Good point! Yes that should be `invalid_abis.erase(enabled_abis[i])`.